### ISSUE TITLE
Python 3.12 build uses distutils

### ISSUE THREAD

**Issue by RaulPPelaez** [conda-forge/mdtraj-feedstock#58]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

The following env brings a package that uses setuptools:
```shell
$ mamba create -n test python==3.12 mdtraj
```
This is causing an error when importing it in the NNPOps feedstock:
```shell
______________________ ERROR collecting TestBatchedNN.py _______________________
ImportError while importing test module '$PREFIX/share/nnpops/tests/TestBatchedNN.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../../lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
TestBatchedNN.py:24: in <module>
    import mdtraj
../../../lib/python3.12/site-packages/mdtraj/__init__.py:30: in <module>
    from .formats.registry import FormatRegistry
../../../lib/python3.12/site-packages/mdtraj/formats/__init__.py:9: in <module>
    from .netcdf import NetCDFTrajectoryFile
../../../lib/python3.12/site-packages/mdtraj/formats/netcdf.py:40: in <module>
    from distutils.version import StrictVersion
E   ModuleNotFoundError: No module named 'distutils'
```

In my system this error does not arise, when digging a little into it I found out this is because python is looking into my system folders for it instead of the conda ones (where it is not present).

### Installed packages

```shell
$ mamba list
# packages in environment at /home/raul/miniforge3/envs/test:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
astunparse                1.6.3              pyhd8ed1ab_0    conda-forge
blosc                     1.21.5               hc2324a3_1    conda-forge
bzip2                     1.0.8                hd590300_5    conda-forge
c-ares                    1.28.1               hd590300_0    conda-forge
c-blosc2                  2.14.4               hb4ffafa_1    conda-forge
ca-certificates           2024.2.2             hbcca054_0    conda-forge
hdf5                      1.14.3          nompi_h4f84152_101    conda-forge
keyutils                  1.6.1                h166bdaf_0    conda-forge
krb5                      1.21.2               h659d440_0    conda-forge
ld_impl_linux-64          2.40                 h55db66e_0    conda-forge
libaec                    1.1.3                h59595ed_0    conda-forge
libblas                   3.9.0           22_linux64_openblas    conda-forge
libcblas                  3.9.0           22_linux64_openblas    conda-forge
libcurl                   8.7.1                hca28451_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 hd590300_2    conda-forge
libexpat                  2.6.2                h59595ed_0    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 13.2.0               h77fa898_7    conda-forge
libgfortran-ng            13.2.0               h69a702a_7    conda-forge
libgfortran5              13.2.0               hca663fb_7    conda-forge
libgomp                   13.2.0               h77fa898_7    conda-forge
liblapack                 3.9.0           22_linux64_openblas    conda-forge
libnghttp2                1.58.0               h47da74e_1    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libopenblas               0.3.27          pthreads_h413a1c8_0    conda-forge
libsqlite                 3.45.3               h2797004_0    conda-forge
libssh2                   1.11.0               h0841786_0    conda-forge
libstdcxx-ng              13.2.0               hc0a3c3a_7    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
lz4-c                     1.9.4                hcb278e6_0    conda-forge
lzo                       2.10              hd590300_1001    conda-forge
mdtraj                    1.9.9           py312hadb1fa1_1    conda-forge
ncurses                   6.5                  h59595ed_0    conda-forge
nomkl                     1.0                  h5ca1d4c_0    conda-forge
numexpr                   2.9.0           py312hed3a10b_100    conda-forge
numpy                     1.26.4          py312heda63a1_0    conda-forge
openssl                   3.3.0                hd590300_0    conda-forge
packaging                 24.0               pyhd8ed1ab_0    conda-forge
pandas                    2.2.2           py312h1d6d2e6_1    conda-forge
pip                       24.0               pyhd8ed1ab_0    conda-forge
py-cpuinfo                9.0.0              pyhd8ed1ab_0    conda-forge
pyparsing                 3.1.2              pyhd8ed1ab_0    conda-forge
pytables                  3.9.2           py312h96d95ec_2    conda-forge
python                    3.12.0          hab00c5b_0_cpython    conda-forge
python-dateutil           2.9.0              pyhd8ed1ab_0    conda-forge
python-tzdata             2024.1             pyhd8ed1ab_0    conda-forge
python_abi                3.12                    4_cp312    conda-forge
pytz                      2024.1             pyhd8ed1ab_0    conda-forge
readline                  8.2                  h8228510_1    conda-forge
scipy                     1.13.0          py312hc2bc53b_1    conda-forge
setuptools                69.5.1             pyhd8ed1ab_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
snappy                    1.2.0                hdb0a2a9_1    conda-forge
tk                        8.6.13          noxft_h4845f30_101    conda-forge
tzdata                    2024a                h0c530f3_0    conda-forge
wheel                     0.43.0             pyhd8ed1ab_1    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
zlib                      1.2.13               hd590300_5    conda-forge
zlib-ng                   2.0.7                h0b41bf4_0    conda-forge
zstd                      1.5.6                ha6fb4c9_0    conda-forge
```

### Environment info

```shell
N/A
```

**Comment by RaulPPelaez** [comment]
The failing NNPOps builds are here: https://github.com/conda-forge/nnpops-feedstock/pull/39

**Comment by mattwthompson** [comment]
This will be gone in the next release https://github.com/mdtraj/mdtraj/pull/1834

I'm fairly surprised we haven't run into this before - the high-level import should be straight-up broken on all 3.12 builds, but we've been shipping them (albeit not on all platforms) for months now cc: @jeremyleung521

**Comment by mattwthompson** [comment]
1.10.0rc1 should fix this, on `conda-forge/label/mdtraj_dev` or something like that

**Comment by jeremyleung521** [comment]
> This will be gone in the next release [mdtraj/mdtraj#1834](https://github.com/mdtraj/mdtraj/pull/1834)
> 
> I'm fairly surprised we haven't run into this before - the high-level import should be straight-up broken on all 3.12 builds, but we've been shipping them (albeit not on all platforms) for months now cc: @jeremyleung521

I had no problems on my end. I tested it out on fresh installs of miniforge etc. and it seems distutils is importable.

I had a quick look and it seems during runtime for `nnpops`, `setuptools`'s downgraded to [65.3.0](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=936195&view=logs&j=385a3385-ae8b-54d1-9abe-3c614f74cb73&t=a5f9d343-149e-5621-6ab9-f0278009be04&l=1407), which was released before python 3.12. So distutils is missing for python 3.12 (now bundled with newer versions of setuptools for py3.12), causing the import issue. I get similar problems when I forcefully downgrade setuptools. This is a nnpops runtime dependency problem in their recipe, but dropping distutils is good practice for mdtraj anyways.

**Comment by mikemhenry** [comment]
>  but dropping distutils is good practice for mdtraj anyways

Agreed, since its dropped in py3.12 and things are only working since `setuptools` decided to vendor `distutils` since it broke so much of the ecosystem. 

> This is a nnpops runtime dependency problem in their recipe

Thank you for tracking that down! I am sure there was a reason for that pin (actually the pin isn't in our feedstock, so it must be a dependency of ours) but I can try and relax it and see if that fixes the problem (and does create a new one).

Oddly `micromamba create -n nnpops "nnpops=0.6" "setuptools==69.5.1"` works just fine so I will try and pin this in our recipe. 

We can keep this issue (IMHO) open until we get a new release

**Comment by mattwthompson** [comment]
I'm offline next week but targeting May 27 as a full release - with any luck some folks will take these RC builds for a spin and nothing huge will break (tracking https://github.com/mdtraj/mdtraj/issues/1860)

**Comment by mikemhenry** [comment]
@mattwthompson that sounds good, as we unjam python 3.12 in the ecosystem we might get more issues, but having the RC out there should fix things until we get the proper release out

**Comment by mattwthompson** [comment]
I'm probably going to make another RC just to add the numpy 1.x pin into PyPI world, but so far people either haven't reported issues or tried the RC1 builds

**Comment by mattwthompson** [comment]
This should be fixed with 1.10.0, but Setuptools vendors `distutils` so most environments won't even know the difference
